lamagraph-compiler/src/Lamagraph/Compiler/Typechecker/Infer/Decl.hs

```diff
@@ -0,0 +1,19 @@
+module Lamagraph.Compiler.Typechecker.Infer.Decl (inferLLmlDecl) where
+
+import Relude
+
+import Control.Lens
+
+import Lamagraph.Compiler.Extension
+import Lamagraph.Compiler.Parser.SrcLoc
+import Lamagraph.Compiler.Syntax
+import Lamagraph.Compiler.Typechecker.Infer.Expr
+import Lamagraph.Compiler.Typechecker.TcTypes
+
+inferLLmlDecl :: TyEnv -> LLmlDecl LmlcPs -> MonadTypecheck (TyEnv, LLmlDecl LmlcTc)
+inferLLmlDecl tyEnv (L loc decl) = over _2 (L loc) <$> inferLmlDecl tyEnv decl
+
+inferLmlDecl :: TyEnv -> LmlDecl LmlcPs -> MonadTypecheck (TyEnv, LmlDecl LmlcTc)
+inferLmlDecl tyEnv = \case
+  ValD _ lBindGroup -> over _2 (ValD noExtField) <$> inferLLmlBindGroup tyEnv lBindGroup
+  _ -> error "Failed to typecheck top-level declaration: unsupported!"
```

Comment:
Can we provide information on what exactly is unsupported?

Reply:
I've tried to just silently skip them, but because of https://gitlab.haskell.org/ghc/ghc/-/issues/15683 I couldn't just use `coerce`. So I made messages a little more clear